hass_nabucasa/files.py

```diff
@@ -0,0 +1,210 @@
+"""Manage cloud files."""
+
+from __future__ import annotations
+
+from collections.abc import AsyncIterator, Callable, Coroutine
+import contextlib
+from json import JSONDecodeError
+import logging
+from typing import TYPE_CHECKING, Any, Literal, TypedDict
+
+from aiohttp import (
+    ClientError,
+    ClientResponse,
+    ClientTimeout,
+    StreamReader,
+    hdrs,
+)
+
+from .auth import CloudError
+
+if TYPE_CHECKING:
+    from . import Cloud, _ClientT
+
+_LOGGER = logging.getLogger(__name__)
+
+_FILE_TRANSFER_TIMEOUT = 43200.0  # 43200s == 12h
+
+type StorageType = Literal["backup"]
+
+
+class FilesError(CloudError):
+    """Exception raised when handling files."""
+
+
+class _FilesHandlerUrlResponse(TypedDict):
+    """URL Response from files handler."""
+
+    url: str
+
+
+class FilesHandlerDownloadDetails(_FilesHandlerUrlResponse):
+    """Download details from files handler."""
+
+
+class FilesHandlerUploadDetails(_FilesHandlerUrlResponse):
+    """Upload details from files handler."""
+
+    headers: dict[str, str]
+
+
+class Files:
+    """Class to help manage files."""
+
+    def __init__(
+        self,
+        cloud: Cloud[_ClientT],
+    ) -> None:
+        """Initialize cloudhooks."""
+        self._cloud = cloud
+
+    def _do_log_response(
+        self,
+        resp: ClientResponse,
+        data: list[Any] | dict[Any, Any] | str | None = None,
+    ) -> None:
+        """Log the response."""
+        isok = resp.status < 400
+        target = (
+            resp.url.path if resp.url.host == self._cloud.servicehandlers_server else ""
+        )
+        _LOGGER.log(
+            logging.DEBUG if isok else logging.WARNING,
```

Comment:
I'd include these details in the exception message and let consumers decide how to log it. Ie always log only at debug level here.

Reply:
Not suited for MVP.